community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -176,6 +178,21 @@ protected void triggered( Void event )
                         lastCommittedTx, lastClosedTx ) );
             }
         };
+        counts.setInitializer( new DataInitializer<CountsAccessor.Updater>()
+        {
+            @Override
+            public void initialize( CountsAccessor.Updater updater )
+            {
+                stringLogger.warn( "Missing counts store, rebuilding it." );
+                new CountsComputer( NeoStore.this ).initialize( updater );
+            }
+
+            @Override
+            public long initialVersion()
+            {
+                return getLastCommittedTransactionId();
```

Comment:
We could, although when we actually invoke this method we are in a state where it wouldn't change, so I actually think it would make the code here unnecessarily complex without providing any value.

It would have been nice if we could fetch the `initialVersion` upon construction of this object and stick it in a final field, but at the time we create this object the `NeoStore` might not have been recovered so it would not be in a consistent enough state to be able to invoke that method.